oio/api/base.py

```diff
@@ -104,6 +108,17 @@ def _direct_request(self, method, url, headers=None, data=None, json=None,
         if self.admin_mode or admin_mode:
             out_headers[ADMIN_HEADER] = '1'
 
+        # Look for a request deadline, deduce the timeout from it.
```

Comment:
Is it possible to export this piece of logic into a method responsible to patch the kwargs?
The `_direct_request()` method is already really long.

Reply:
I'll think about it.